backend/s3/s3.go

```diff
@@ -2137,7 +2150,50 @@ func (f *Fs) bucketExists(ctx context.Context, bucket string) (bool, error) {
 // Mkdir creates the bucket if it doesn't exist
 func (f *Fs) Mkdir(ctx context.Context, dir string) error {
 	bucket, _ := f.split(dir)
-	return f.makeBucket(ctx, bucket)
+	e := f.makeBucket(ctx, bucket)
+	if e != nil {
+		return e
+	}
+	if f.opt.DirectoryMarkers {
+		return f.uploadDirFile(ctx, bucket, dir)
+	}
+	return nil
+}
+
+// Creates empty file as directory marker
+func (f *Fs) uploadDirFile(ctx context.Context, bucket string, dir string) (err error) {
+	dirFilePath := fmt.Sprintf("%s/", dir)
+	fs.Debugf(nil, "Creating directory object %v", dirFilePath)
+	req := s3.PutObjectInput{
```

Comment:
Unfortunate duplication here...

Can you not just call Object.Update with a constructed Object?

Reply:
This required o.split change to reattach trailing slash and an extra metadata header; Done